src/utilities/findBreakingChanges.js

```diff
@@ -395,6 +396,9 @@ function findArgChanges(
           description: `${oldType.name}.${oldField.name} arg ${oldArg.name} defaultValue was removed.`,
         });
       } else {
+        // Since we looking only for client's observable changes we should
+        // compare default values in the same representation as they are
+        // represented inside introspection.
```

Comment:
> For input type default values we apparently just compare it's stringified value instead of doing a deep comparison.

@Neitsch It's intentional since we want to detect only changes observable by the client.
So we should compare `defaultValues` in the same representation as used by introspection:
https://github.com/graphql/graphql-js/blob/0d2220f0a64238473f4e1c9aa8f73f891d0fd3e0/src/type/introspection.js#L337-L345

It's important because `defaultValue` is always specified in "internal" format (same format as returned by `parseLiteral`/`parseValue` and passed into `serialize`) that mean it can be for example instances of `Date` class and `===` wouldn't work for them but if you pass both values into `serialize` you can compare resulting strings without any problems.